### PR TITLE
Fix MSBuildWorkspace Test when run on local machine

### LINKED ISSUE #50258: It looks like Microsoft.CodeAnalysis.MSBuild.UnitTests.MSBuildWorkspaceTests.TestOpenProject_AssemblyNameIsPath unit-test is flaky
Lately, I am consistently getting the test failure when run it on my dev box. Both from command line and from VS:
```
Microsoft.CodeAnalysis.MSBuild.UnitTests.MSBuildWorkspaceTests.TestOpenProject_AssemblyNameIsPath [FAIL]
  Assert.Equal() Failure
                                   ↓ (pos 50)
  Expected: ···l\Temp\2\RoslynTests
  Actual:   ···l\Temp\2\RoslynTests\fc082681-b2ee-4795-a7b1-cda1d73c7e31\CSh···
                                   ↑ (pos 50)
  Stack Trace:
    F:\GitHub\roslyn\src\Workspaces\MSBuildTest\MSBuildWorkspaceTests.cs(830,0): at Microsoft.CodeAnalysis.MSBuild.UnitTests.MSBuildWorkspaceTests.<TestOpenProject_AssemblyNameIsPath>d__40.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

The guid is different from run to run.
However, CI for a GitHub PR is passing.

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes #50258

### CODE REVIEW

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs [hunk 1]
```diff
@@ -826,7 +826,7 @@ public async Task TestOpenProject_AssemblyNameIsPath()
                 var project = solution.Projects.First();
                 var comp = await project.GetCompilationAsync();
                 Assert.Equal("ReproApp", comp.AssemblyName);
-                var expectedOutputPath = GetParentDirOfParentDirOfContainingDir(project.FilePath);
+                var expectedOutputPath = Path.GetDirectoryName(project.FilePath);
```
Comment (jasonmalinowski) [inline src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:829]: Looks like GetParentDirOfParentDirOfContainingDir can be deleted now?

**Comment by JoeRobich** [comment]
This fix is necessary for the upcoming switch to using 16.9 Preview 3 images for our integration CI. Perhaps we should force integration CI to run on the scouting image by hardcoding the queue to 'BuildPool.Windows.VS2019.Pre.Scouting.Open' and opening a tracking issue to revert the line once the non-scounting image is updated.

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs (hunk 1 above)
Comment (jmarolf) [reply src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs]: done

**Comment by jmarolf** [comment]
@jasonmalinowski I removed the test and took all the "info" suggestions for the file.

**Comment by jasonmalinowski** [comment]
@jmarolf We were continuing to chat in the Teams conversation and proposed to keep the test since it's now sounding like this is expected behavior, is that best to do now?

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs [hunk 2]
```diff
@@ -3012,24 +2755,22 @@ public async Task MSBuildProjectShouldHandleDefaultCodePageProperty()
...
+            var text = await document.GetTextAsync();
+            Assert.Equal(new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true), text.Encoding);
+            Assert.Equal("//\u201C", text.ToString());
         }
 
         [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
         [WorkItem(981208, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/981208")]
         [WorkItem(28639, "https://github.com/dotnet/roslyn/issues/28639")]
-        public void DisposeMSBuildWorkspaceAndServicesCollected()
+        public DisposeMSBuildWorkspaceAndServicesCollected()
```
Comment (JoeRobich) [inline src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:2769]:
```suggestion
        public void DisposeMSBuildWorkspaceAndServicesCollected()
```

**Comment by jmarolf** [comment]
@jasonmalinowski alright, added the test back conditionally

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs (hunk 2 above)
Comment (JoeRobich) [reply src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:2788]: @jmarolf  The return type is still missing

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs [hunk 3]
```diff
@@ -3043,76 +2803,30 @@ public void DisposeMSBuildWorkspaceAndServicesCollected()
...
-            project.ReleaseStrongReference();
-            workspace.UseReference(static w => w.Dispose());
-
-            compilation.AssertReleased();
-            sol.AssertReleased();
-        }
-
-        [ConditionalFact(typeof(VisualStudioMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
-        [WorkItem(1088127, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/1088127")]
-        public async Task MSBuildWorkspacePreservesEncoding()
```
Comment (jasonmalinowski) [inline src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:3057]: Where did this go?

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs [hunk 4]
```diff
@@ -908,6 +811,25 @@ public void TestOpenProject_ProjectFileExtensionAssociatedWithUnknownLanguage()
             });
         }
 
+        [ConditionalFact(typeof(VisualStudio16_9_Preview3OrHigherMSBuildInstalled)), Trait(Traits.Feature, Traits.Features.MSBuildWorkspace)]
+        [WorkItem(531631, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/531631")]
+        public async Task TestOpenProject_AssemblyNameIsPath()
```
Comment (jasonmalinowski) [inline src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:816]: Why the move? It just means you're going to get blamed for this in the future.

File: src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs (hunk 3 above)
Comment (JoeRobich) [reply src/Workspaces/MSBuildTest/MSBuildWorkspaceTests.cs:3057]: Line 3047  - Line 3088 from the earlier version of this file seem to be missing.

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**